Applications/Spire/Source/Ui/DropDownBox.cpp

```diff
@@ -0,0 +1,249 @@
+#include "Spire/Ui/DropDownBox.hpp"
+#include <QCoreApplication>
+#include <QHBoxLayout>
+#include <QKeyEvent>
+#include <QVBoxLayout>
+#include "Spire/Spire/Dimensions.hpp"
+#include "Spire/Ui/ArrayListModel.hpp"
+#include "Spire/Ui/Button.hpp"
+#include "Spire/Ui/CustomQtVariants.hpp"
+#include "Spire/Ui/DropDownList.hpp"
+#include "Spire/Ui/Icon.hpp"
+#include "Spire/Ui/LayeredWidget.hpp"
+#include "Spire/Ui/ListItem.hpp"
+#include "Spire/Ui/ListView.hpp"
+#include "Spire/Ui/TextBox.hpp"
+
+using namespace boost::signals2;
+using namespace Spire;
+using namespace Spire::Styles;
+
+namespace {
+  auto DEFAULT_STYLE() {
+    auto style = StyleSheet();
+    style.get(ReadOnly() >> is_a<Icon>()).
+      set(Visibility::INVISIBLE);
+    style.get(ReadOnly() >> is_a<Button>()).
+      set(Visibility::INVISIBLE);
+    style.get(Disabled() >> is_a<Icon>()).
+      set(Fill(QColor::fromRgb(0xC8, 0xC8, 0xC8)));
+    return style;
+  }
+
+  auto ICON_STYLE(StyleSheet style) {
+    style.get(Any()).
+      set(BackgroundColor(QColor::fromRgb(0, 0, 0, 0))).
+      set(Fill(QColor::fromRgb(0x33, 0x33, 0x33)));
+    return style;
+  }
+
+  auto TEXT_BOX_STYLE(StyleSheet style) {
+    style.get(Any()).
+      set(PaddingRight(scale_width(22)));
+    style.get(ReadOnly()).
+      set(PaddingRight(scale_width(0)));
+    return style;
+  }
+}
+
+class DropDownBox::DropDownListWrapper : public QWidget {
+  public:
+    DropDownListWrapper(ListView& list_view, QWidget* parent)
+        : QWidget(parent),
+          m_list_view(&list_view) {
+      m_drop_down_list = new DropDownList(*m_list_view, parent);
+      m_panel = m_drop_down_list->window();
+      m_drop_down_list->setFocusProxy(m_list_view);
+      auto list_model = m_list_view->get_list_model();
+      for(auto i = 0; i < list_model->get_size(); ++i) {
+        m_list_view->get_list_item(i)->setFocusPolicy(Qt::NoFocus);
+      }
+      m_list_view->installEventFilter(this);
+      m_panel->installEventFilter(this);
+    }
+
+    QWidget* get_panel() const {
```

Comment:
If the panel can not be ```nullptr``` then return a reference. Pointers are used for non-owning nullable references.
The exception to this is that fields are never references.